hyperspy/misc/test_utils.py

```diff
@@ -163,6 +168,52 @@ def assert_warns(message=None, category=None):
             raise ValueError(msg)
 
 
+def get_matplotlib_version_label():
+    """ Return a string describing the matplotlib version installed. The string
+    is used to label the references images for plot testing. """
+    version = LooseVersion(matplotlib.__version__)
+    if version >= LooseVersion('2.0.0'):
+        return 'mpl2'
+    else:
+        return 'mpl1'
+
+
+def reset_rcParams_default():
+    import matplotlib.pyplot as plt
+    plt.rcParams.clear()
+    plt.rcParams.update(plt.rcParamsDefault)
+
+
+@simple_decorator
+def switch_backend_mpl(function):
```

Comment:
Before this decorator was used a bit more with nose. Now with pytest, it is used only once, so it doesn't make sense anymore to have it as a decorator...

This is interesting to know how it's possible to set the backend in travis, however, I don't think this is necessary because this [function](https://github.com/ericpre/hyperspy/blob/testing_plot/hyperspy/misc/test_utils.py#L181) is called in __init__.py and reset rcParams to default values, which set the backend to `agg` at the same time. It also reset any user rcParams and should make plot testing more reliable.